### ISSUE TITLE
Missing symbolic link to data run (Trac #2201)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2193]
Hi,

there is a symbolic link missing for run 00131184.
/data/exp/IceCube/2018/filtered/level2/0619/Run00131184_73
does exist but
/data/exp/IceCube/2018/filtered/level2/0619/Run00131184
does not.

That would not be too annoying but the goodRun file references the /data/exp/IceCube/2018/filtered/level2/0619/Run00131184 folder. 
As the other folders for 2018 seem to be mostly symlinked, I would like this folder to be symlinked as well.

Cheers,
Frederik

Migrated from https://code.icecube.wisc.edu/ticket/2201
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:18:07", 
    "description": "Hi,\n\nthere is a symbolic link missing for run 00131184.\n/data/exp/IceCube/2018/filtered/level2/0619/Run00131184_73\ndoes exist but\n/data/exp/IceCube/2018/filtered/level2/0619/Run00131184\ndoes not.\n\nThat would not be too annoying but the goodRun file references the /data/exp/IceCube/2018/filtered/level2/0619/Run00131184 folder. \nAs the other folders for 2018 seem to be mostly symlinked, I would like this folder to be symlinked as well.\n\nCheers,\nFrederik\n\n", 
    "reporter": "flauber", 
    "cc": "", 
    "resolution": "wontfix", 
    "_ts": "1550067487593768", 
    "component": "infrastructure", 
    "summary": "Missing symbolic link to data run", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-10-29T17:24:29", 
    "milestone": "", 
    "owner": "rsnihur", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20181029T18:17:49: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "rsnihur"

**Comment by icecube-trac** [comment]
Trac update at 20181030T21:33:05: rsnihur commented:

> Hi Frederik,
> This is a test run so symbolic links are explicitly NOT created.
> Thx,
> -Rob
>

**Comment by icecube-trac** [comment]
Trac update at 20181031T10:41:54: flauber commented:

> Hi,
> 
> so the error is actually the run showing up in the IC86_2018_GoodRunInfo.txt?
>

**Comment by icecube-trac** [comment]
Trac update at 20181031T15:34:46: rsnihur commented:

> Hi Frederik,
> Below I cut-and-pasted 2 discussions about this issue from the Slack
> #data-processing channel dated June 25-26 & September 11.
> 
> AFAIK the test runs should be in IC86_2018_GoodRunInfo.txt 
> 
> Thx,
> -Rob
> 
> 
> -----
> 
> flambda [12:21 PM]
> btw, will the IC86_2018_GoodRunInfo.txt be fixed? It actually does point to non existent folders (my guess is someone forgot to add the symlinks to the _73 folders)
> 
> flambda [8:12 AM]
> and while we are at it, should the 2017 test runs also show up in the IC86_2017_GoodRunInfo? I have been pooling these files for infos on hte runs to not manually walk over the fs, is there a better way to do that?
> 
> Rob Snihur [9:06 AM]
> I can take a look at the IC86_2018_GoodRunInfo.txt , but should the test runs actually be in there?  Especially if there are no test runs in the IC86_2017_GoodRunInfo.txt ...
> 
> dschultz [9:10 AM]
> I think if the test runs are marked as GOOD they should be in the good run list.  for 2017, I think they should be present.  for 2018, they probably need to be approved first
> 
> jkelley [9:24 AM]
> FYI, the 2017 test runs (129393–129397) are marked BAD.
> As @dschultz says, good test runs should be included in the good run list the year they are the test for
> 
> flambda [9:36 AM]
> hm, so what is the canonical way to poll information about runs? Especially since we have pass2 and if I also want to know there the data is stored? In the past I just got me a local copy of the https://live.icecube.wisc.edu/snapshots but that afaik does not give the path to the data (also, maybe I should move this to software)
> 
> jkelley [9:42 AM]
> As far as I know the good run list file is intended to be the canonical way to provide paths to the L2 data
> 
> flambda [9:44 AM]
> kk, thanks
> 
> -------
> 
> 
> 
> thomas [4:46 AM]
> Hi!
> I was looking for 2018 Level2 data and cannot find it… The folders in `/data/exp/IceCube/2018/filtered/level2` stop on July 8th, the date of the switch to 2018 data-taking. Also, the corresponding GRL only mentions the 2018 test runs (`/data/exp/IceCube/2018/filtered/level2/IC86_2018_GoodRunInfo.txt`). (edited)
> Also, the symlinks for the runs (e.g. `Run00131185_73/` to `Run00131185/`) are missing.
> 
> nick [6:05 AM]
> joined #data-processing.
> 
> dschultz [8:02 AM]
> they're not processed yet
> I think 2018 L2 will be starting soon, and you'll hear more about it at the collaboration meeting
> 
> Rob Snihur [9:38 AM]
> Hi @thomas I am working on it
> 
> Rob Snihur [11:40 AM]
> @thomas NB: The symlinks are purposely not created for test runs (131185 is a test run)
> 
> thomas [11:40 AM]
> thanks for looking into it, Rob!
> but the test runs are mentioned in the GoodRunInfo.txt file and the entry there points to the (missing) symlink (edited)
> 
> Rob Snihur [11:45 AM]
> I'm not sure why the GoodRunInfo.txt file points to the missing symlink.  Can you check if that's how it also was for the 2017 season?  I know that the L2 processing scripts explicitly do not create the links for test runs
> 
> thomas [4:10 PM]
> The last time that the test runs were included was in 2015, the GoodRunInfo.txt back then points to the symlinks (which exist).
> I also checked the GoodRunInfo_Versioned.txt, which should point to the correct folders, but it points to e.g. `Run00131184_74` when the folder is actually `Run00131184_73`.
> For 2016 and 2017, the symlinks are not there, but the runs also do not show up in the GoodRunInfo.txt. I think this might be the actual issue, as the entries are also incomplete (having `N/A` for `ActiveStrings`, etc.). There is one more run in the file which is not a test run, and everything looks good there.
> I just noticed this because my parsing script from the previous years failed on the 2018 GoodRunInfo.txt file.

**Comment by icecube-trac** [comment]
Trac update at 20181031T16:18:29: flauber commented:

> Yeah, I had the same problem as thomas (parsing script works for all years but 2018).
> Going though all years, the only runs without symlinks are these here:
> 
> /data/exp/IceCube/2018/filtered/level2/0619/Run00131184
> /data/exp/IceCube/2018/filtered/level2/0620/Run00131185
> /data/exp/IceCube/2018/filtered/level2/0620/Run00131191
> /data/exp/IceCube/2018/filtered/level2/0620/Run00131192
> /data/exp/IceCube/2018/filtered/level2/0620/Run00131193
> /data/exp/IceCube/2018/filtered/level2/0620/Run00131194
> which are the test runs for 2018.
> 
> I havent found a list of all test runs for all the years but for example in 2012,
> the second test runs were  Runs 120028-120030. They are in the GoodRunList (also with string information and the like which is missing for 2018) as well as symlinked.
> So in my eyes, these runs are either good and hence in the GoodRunlist with all their information and hence also symlinked or they are bad and hence not in the GoodRunList and hence also not symlinked.
> 
> If you want to experiment, this was my script to check the folders:
> from __future__ import division, absolute_import, with_statement, print_function#, unicode_literals
> import os
> import collections
> 
> 
> 
> pass2dataset = collections.namedtuple('Pass2Dataset',
>                                       ['run', 'good_i3', 'good_it', 'livetime', 'outdir', 'active_strings',
>                                        'active_doms', 'active_inice', 'year'])
> 
> 
> 
> 
> def pass2_dataruns(only_burnsample=True, years=("2016", ), _good_i3=True):
>     basefolder = "/data/exp/IceCube"
>     pass2datasets = []
>     for year in years:
>         good_run_file = os.path.join(basefolder, year, ("filtered/level2pass2" if int(year) <= 2016 else "filtered/level2"), "IC86_" + year + "_GoodRunInfo.txt")
>         with open(good_run_file) as infile:
>             for line in infile:
>                 if line.startswith("RunNum") or line.strip().startswith("(1=good 0=bad)"):
>                     continue
>                 a = line.split()
>                 run = a[0]
>                 if only_burnsample and run[-1] != "0":
>                     continue
>                 run = int(run)
>                 good_i3 = bool(int(a[1]))
>                 if _good_i3 is not None and good_i3 != _good_i3:
>                     continue
>                 good_it = bool(int(a[2]))
>                 livetime = float(a[3])
>                 active_strings = int(a[4]) if not a[4] == "N/A" else -1
>                 active_doms = int(a[5]) if not a[5] == "N/A" else -1
>                 active_inice = int(a[6]) if not a[6] == "N/A" else -1
>                 outdir = a[7]
> 
>                 d = pass2dataset(run, good_i3, good_it,
>                                      livetime, outdir,
>                                      active_strings,
>                                      active_doms,
>                                      active_inice,
>                                      int(year))
>                 pass2datasets.append(d)
>     return pass2datasets
> 
> if __name__ == "__main__":
>     pass2datasets = pass2_dataruns(years=("2012", "2013", "2014", "2015", "2016", "2017", "2018",), only_burnsample=False)
> 
>     for pass2dataset in pass2datasets:
>         outdir = pass2dataset.outdir
>         if not os.path.isdir(outdir):
>             print(outdir)
>         #os.path.islink(outdir)):
> 
> 
> 
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20181031T16:59:57: rsnihur commented:

> Hi Frederik,
> 
> As I said, the script 
> /data/user/i3filter/IC86_OfflineProcessing/OfflineSubmitScripts_2018/PostProcessing_L2.py
> has the following lines:
> 
>     if not run.is_test_run():
>         logger.info('Create run sym link')
>         make_relative_symlink(run.format(config.get_l2_path_pattern(run.get_season(), 'RUN_FOLDER')), run.format(config.get_l2_path_pattern(run.get_season(), 'RUN_FOLDER_LINK')), args.dryrun, logger, replace = True)
> 
> 
> This is the same script that was used for season 2017.
> I hesitate to make the symlinks manually for 2018 test runs because it may cause other unforeseen problems for other people.
> 
> Perhaps the 2018 test runs should be marked bad instead.
> I do not make those decisions: it's up to the detector operations group,
> and John Kelley has already stated his opinion on Slack.
> 
> Thx,
> -Rob

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:18:07: 
* olivas changed status from "assigned" to "closed"
* olivas changed resolution from "" to "wontfix"
